### PR TITLE
[webview_flutter] Support for handling basic authentication requests (Platform Interface)

### PR DESCRIPTION
**Pull request by JeroenWeener**
Adds the platform interface implementation for basic http authentication.

This PR is part of a series of PRs that aim to close https://github.com/flutter/flutter/issues/83556.
The PR that contains all changes can be found at https://github.com/flutter/packages/pull/4140.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by JeroenWeener** [comment]
@stuartmorgan thank you for noticing. Turns out that new files are not removed when checking out the `main` branch. I removed them manually, leaving only the platform interface changes.

**Comment by stuartmorgan** [comment]
> Turns out that new files are not removed when checking out the `main` branch.

Oh, right 🤦🏻  Sorry about that, apparently I haven't run into the new files case recently so I forgot the workflow I described wouldn't handle that.

File: packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/webview_credential.dart [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import 'package:meta/meta.dart';
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/webview_credential.dart:1]: It looks like we missed this in the other PR. This was missing a license.

```suggestion
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:meta/meta.dart';
```

File: packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/http_auth_request.dart [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+///     return WKWebViewHttpAuthRequest._(
+///       authRequest,
+///       extraData: extraData,
+///     );
+///   }
+///
+///   final String? extraData;
+/// }
+/// ```
+/// {@end-tool}
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/http_auth_request.dart:48]: nit: 
```suggestion
```

File: packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/http_auth_request.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import 'webview_credential.dart';
+
+/// Defines the parameters of a pending HTTP authentication request received by
+/// the webview through a [HttpAuthRequestCallback].
+///
+/// Platform specific implementations can add additional fields by extending
+/// this class and providing a factory method that takes the [HttpAuthRequest]
+/// as a parameter.
+///
+/// {@tool sample}
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/lib/src/types/http_auth_request.dart:15]: nit:
```suggestion
```

File: packages/webview_flutter/webview_flutter_platform_interface/test/platform_navigation_delegate_test.dart [hunk 4]
```diff
@@ -142,6 +142,19 @@ void main() {
       throwsUnimplementedError,
     );
   });
+
+  test(
+      'Default implementation of setOnReceiveHttpAuthRequest should throw unimplemented error',
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter_platform_interface/test/platform_navigation_delegate_test.dart:147]: nit: 
```suggestion
      'Default implementation of setOnHttpAuthRequest should throw unimplemented error',
```

**Review by bparrishMines — APPROVED** [review]
LGTM with a few nits

cc @stuartmorgan for secondary review

**Comment by JeroenWeener** [comment]
Thanks @bparrishMines, I updated the branch based on your comments.

File: packages/webview_flutter/webview_flutter_platform_interface/pubspec.yaml [hunk 5]
```diff
@@ -14,7 +14,7 @@ dependencies:
   flutter:
     sdk: flutter
   meta: ^1.7.0
-  plugin_platform_interface: ^2.1.0
+  plugin_platform_interface: ^2.1.6
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter_platform_interface/pubspec.yaml:17]: Why is this changing? I don't think anything changed in that version that we could be depending on.

**Review by stuartmorgan — APPROVED** [review]
LGTM with one question.

**Comment by stuartmorgan** [comment]
Overriding the out-of-date tree check; the dashboard is for some reason showing that run 490 is the last run of the failing test, but GitHub UI is correctly registering [run 491](https://ci.chromium.org/ui/p/flutter/builders/prod/Linux_android%20android_platform_tests_shard_1%20stable/491/overview) which is a re-run that passed.

**Pull request merged**